File: mobile/lib/models/decoration_model.dart
class DecorationModel {
  final int id;
  final int sessionId;
  final String type; // sticker | text | emoji
  final String content;
  final double posX;
  final double posY;
  final double scale;
  final double rotation;

  DecorationModel({
    required this.id,
    required this.sessionId,
    required this.type,
    required this.content,
    this.posX = 0,
    this.posY = 0,
    this.scale = 1,
    this.rotation = 0,
  });

  factory DecorationModel.fromJson(Map<String, dynamic> json) {
    return DecorationModel(
      id: json['id'],
      sessionId: json['session_id'],
      type: json['type'] ?? 'sticker',
      content: json['content'],
      posX: (json['pos_x'] ?? 0).toDouble(),
      posY: (json['pos_y'] ?? 0).toDouble(),
      scale: (json['scale'] ?? 1).toDouble(),
      rotation: (json['rotation'] ?? 0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() => {
        'type': type,
        'content': content,
        'pos_x': posX,
        'pos_y': posY,
        'scale': scale,
        'rotation': rotation,
      };
}

File: backend/src/models/decorationModel.js
const db = require('../config/database');

const DecorationModel = {
  create({ session_id, type, content, pos_x, pos_y, scale, rotation }) {
    const info = db
      .prepare(
        `INSERT INTO decorations (session_id, type, content, pos_x, pos_y, scale, rotation)
         VALUES (?, ?, ?, ?, ?, ?, ?)`
      )
      .run(
        session_id,
        type || 'sticker',
        content,
        pos_x ?? 0,
        pos_y ?? 0,
        scale ?? 1,
        rotation ?? 0
      );
    return this.findById(info.lastInsertRowid);
  },

  findById(id) {
    return db.prepare(`SELECT * FROM decorations WHERE id = ?`).get(id);
  },

  findBySession(session_id) {
    return db.prepare(`SELECT * FROM decorations WHERE session_id = ?`).all(session_id);
  },

  update(id, { pos_x, pos_y, scale, rotation, content }) {
    db.prepare(
      `UPDATE decorations SET
        pos_x = COALESCE(?, pos_x),
        pos_y = COALESCE(?, pos_y),
        scale = COALESCE(?, scale),
        rotation = COALESCE(?, rotation),
        content = COALESCE(?, content)
       WHERE id = ?`
    ).run(pos_x, pos_y, scale, rotation, content, id);
    return this.findById(id);
  },

  remove(id) {
    db.prepare(`DELETE FROM decorations WHERE id = ?`).run(id);
    return true;
  },
};

module.exports = DecorationModel;
